packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts

```diff
@@ -0,0 +1,40 @@
+import type { PersonaInformation, NextIDPlatform } from '@masknet/shared-base'
+import { useAsync } from 'react-use'
+import { useMemo, useCallback } from 'react'
+import Services from '../../extension/service'
+import { activatedSocialNetworkUI } from '../../social-network'
+import { useLastRecognizedIdentity } from './useActivatedUI'
+import { useMyPersonas } from './useMyPersonas'
+import { useSetupGuideStatusState } from './useNextID'
+import { PersonaContext } from '../../extension/popups/pages/Personas/hooks/usePersonaContext'
+
+export function usePersonaAgainstSNSConnectStatus() {
+    const ui = activatedSocialNetworkUI
+    const platform = ui.configuration.nextIDConfig?.platform as NextIDPlatform
+    const lastState = useSetupGuideStatusState()
+    const { currentPersona } = PersonaContext.useContainer()
+    const lastRecognized = useLastRecognizedIdentity()
+    const username = useMemo(() => {
+        return lastState.username || lastRecognized.identifier?.userId
+    }, [lastState, lastRecognized])
+    const personas = useMyPersonas()
+
+    const checkSNSConnectToCurrentPersona = useCallback(
```

Comment:
This function is only used in the async function. Why you need to memo it? Please move it into the async function.